frontend-ssr/components/cli_git/GitCli_CC.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { uuid } from '../../services/Utils';
+
+import styles from './GitCli.module.scss';
+
+interface Props {
+  commands: RegExp[] | string[],
+  successMessages: string[],
+}
+
+interface State {
+  currentIndex: number,
+  messages: Message[],
+  btnMessage: string,
+  loading: boolean,
+}
+
+interface Message {
+  text: string,
+  type: MessageType,
+  time: number,
+  id: string,
+}
+
+enum MessageType {
+  COMMAND = 'command',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+class matchCommandMessageClassComponent extends React.Component<Props, State, Message> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentIndex: 0,
+      messages: [],
+      btnMessage: 'Next',
+      loading: false,
+    };
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    const formEl = event.target;
+    const gitCommand = event.target.gitCommand.value;
+    const { commands, successMessages } = this.props;
+    const { currentIndex } = this.state;
+
+    let isMatch = false;
+
+    if (commands[currentIndex] instanceof RegExp) {
+      isMatch = (commands[currentIndex] as RegExp).test(gitCommand);
+    } else {
+      isMatch = commands[currentIndex] === gitCommand.trim().toLowerCase();
+    }
+
+    const commandMessage: Message = {
+      text: gitCommand,
+      type: MessageType.COMMAND,
+      time: Date.now(),
+      id: uuid(),
+    };
+
+    let matchCommandMessage: Message;
+
+    if (!isMatch) {
+      matchCommandMessage = {
+        text: 'This is not a valid command',
+        type: MessageType.ERROR,
+        time: Date.now(),
+        id: uuid(),
+      };
+    } else {
+      matchCommandMessage = {
+        text: successMessages[currentIndex],
+        type: MessageType.SUCCESS,
+        time: Date.now(),
+        id: uuid(),
+      };
+    }
+
+    if (!isMatch) {
+      this.setState(({ messages }) => ({
```

Comment:
A little bit of duplicate code between those if statements. I think we can simplify them a lot.

First thing I notice is that no matter the case, we always insert the command and the command message into the `messages` Array. So let's move that state change outside the if statements.

```
this.setState(({ messages }) => ({
   messages: [...messages, commandMessage, matchCommandMessage],
   currentIndex: isMatch ? currentIndex + 1: currentIndex
}));
```

After that, I think we only need one `if statement`, to check if we need to change the button text

```
if (isMatch && currentIndex + 2 === commands.length) {
  // One before the end
  this.setState({
    btnMessage: 'Done'
  })
}
```